internal/service/componentdescriptor/untar_internal_test.go

```diff
@@ -0,0 +1,121 @@
+package componentdescriptor
+
+import (
+	"archive/tar"
+	"bytes"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnTar(t *testing.T) {
```

Comment:
The `maxSize = maxDescriptorSizeBytes` path is uncovered. Not sure if worth testing, just wanted to mention it here

<img width="421" height="164" alt="image" src="https://github.com/user-attachments/assets/26ae3fd1-b17d-4cda-bcd3-2f9ede943981" />